File: backend/CncApp/CncApp.Application/Dtos/Jobs/JobReportDto.cs
namespace CncApp.Application.Dtos.Jobs;

public class JobReportDto
{
    // Job context
    public int Id { get; set; }
    public int OrderId { get; set; }
    public string MachineName { get; set; } = string.Empty;
    public string PartName { get; set; } = string.Empty;
    public string PartNumber { get; set; } = string.Empty;
    public DateOnly DueDate { get; set; }
    public DateTimeOffset? StartedDateTime { get; set; }
    public DateTimeOffset? EndedDateTime { get; set; }
    public string JobStatus { get; set; } = string.Empty;

    // Job totals (aggregated from shifts)
    public int PartAmountPlanned { get; set; }
    public int TotalPartsMade { get; set; }
    public int TotalScrap { get; set; }
    public int BarAmountPlanned { get; set; }
    public int TotalBarsConsumed { get; set; }
    public int? EstimatedPartsPerBar { get; set; }
    public decimal? ActualPartsPerBar { get; set; }
    public TimeSpan TotalDowntime { get; set; }

    // Shift history
    public List<JobReportShiftDto> Shifts { get; set; } = new();
}

File: frontend/angular/src/app/core/dtos/jobs/job-report.dto.ts
export interface JobReportIssueLogDto {
  id: number;
  shiftId: number;
  operatorName: string;
  createdDateTime: string;
  issueType: number;
  description: string;
  scrapQuantity: number;
  downtime: string | null;
}

export interface JobReportShiftDto {
  id: number;
  operatorName: string;
  startTime: string;
  stopTime: string | null;
  partsMade: number;
  scrap: number;
  barsConsumed: number;
  partsPerBar: number | null;
  downtime: string | null;
}

export interface JobReportDto {
  id: number;
  orderId: number;
  machineName: string;
  partName: string;
  partNumber: string;
  dueDate: string;
  startedDateTime: string | null;
  endedDateTime: string | null;
  jobStatus: string;
  partAmountPlanned: number;
  totalPartsMade: number;
  totalScrap: number;
  barAmountPlanned: number;
  totalBarsConsumed: number;
  estimatedPartsPerBar: number | null;
  actualPartsPerBar: number | null;
  totalDowntime: string;
  totalUptime: string;
  shifts: JobReportShiftDto[];
  issueLogs: JobReportIssueLogDto[];
}
